tools/test/LcovMerger/java/com/google/devtools/lcovmerger/LcovMergerFlags.java

```diff
@@ -0,0 +1,84 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.lcovmerger;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+@AutoValue
+abstract class LcovMergerFlags {
+
+    @Nullable
+    abstract String coverageDir();
+    @Nullable
+    abstract String reportsFile();
+    abstract String outputFile();
+    abstract List<String> filterSources();
+
+
+    /**
+     * Parse flags in the form of "--coverage_dir=... -output_file=..."
+     */
+    static LcovMergerFlags parseFlags(String[] args) {
+      ImmutableList.Builder<String> filterSources = new ImmutableList.Builder<>();
+      String coverageDir = null;
+      String reportsFile = null;
+      String outputFile = null;
+
+      for (String arg : args) {
+        if (!arg.startsWith("--")) {
+          throw new IllegalArgumentException("Argument (" + arg + ") should start with --");
+        }
+        String[] parts = arg.substring(2).split("=", 2);
+        if (parts.length != 2) {
+           throw new IllegalArgumentException("There should be = in argument (" + arg + ")");
+        }
+        switch (parts[0]) {
+          case "coverage_dir":
+            coverageDir = parts[1];
+            break;
+          case "reports_file":
+            reportsFile = parts[1];
+            break;
+          case "output_file":
+            outputFile = parts[1];
+            break;
+          case "filter_sources":
+              filterSources.add(parts[1]);
+            break;
+          default:
+            throw new IllegalArgumentException("Unknown flag --" + parts[0]);
```

Comment:
print the whole flag (`arg`) to give more context in the error

Reply:
Done.